clients/fides-js/src/lib/consent-utils.ts

```diff
@@ -149,17 +149,21 @@ export const experienceIsValid = (
     return false;
   }
   if (
-    effectiveExperience.component !== ComponentType.OVERLAY &&
-    effectiveExperience.component !== ComponentType.TCF_OVERLAY
+    effectiveExperience.experience_config?.component !== ComponentType.MODAL &&
+    effectiveExperience.experience_config?.component !==
+      ComponentType.BANNER_AND_MODAL &&
+    effectiveExperience.experience_config?.component !==
+      ComponentType.TCF_OVERLAY
```

Comment:
nit: this kind of logic is really hard to read so I'd extract into a local:
```
const componentType = effectiveExperience.experience_config?.component;
if (componentType !== ...
```

Reply:
removed this anyway as we already have an API filter for this